packages/compat/src/compat-app-builder.ts

```diff
@@ -855,19 +837,19 @@ export class CompatAppBuilder {
   private gatherAssets(inputPaths: OutputPaths<TreeNames>): Asset[] {
     // first gather all the assets out of addons
     let assets: Asset[] = [];
-    for (let pkg of this.allActiveAddons) {
-      if (pkg.meta['public-assets']) {
-        for (let [filename, appRelativeURL] of Object.entries(pkg.meta['public-assets'] || {})) {
-          let sourcePath = resolvePath(pkg.root, filename);
-          let stats = statSync(sourcePath);
-          assets.push({
-            kind: 'on-disk',
-            sourcePath,
-            relativePath: appRelativeURL,
-            mtime: stats.mtimeMs,
-            size: stats.size,
-          });
-        }
+
+    let pkg = this.allActiveAddons.find(pkg => pkg.name === '@embroider/synthesized-styles');
+    if (pkg?.meta['public-assets']) {
+      for (let [filename, appRelativeURL] of Object.entries(pkg.meta['public-assets'] || {})) {
+        let sourcePath = resolvePath(pkg.root, filename);
+        let stats = statSync(sourcePath);
+        assets.push({
+          kind: 'on-disk',
+          sourcePath,
+          relativePath: appRelativeURL,
+          mtime: stats.mtimeMs,
+          size: stats.size,
+        });
```

Comment:
I would have deleted this whole block so we no longer generate the public-assets coming from addons in the rewritten-app. 

However, it seems the CSS of the app itself (`dist/assets/my-app.css`) is generated here out of the `@embroider/synthesized-styles` package. So I left the handling we had before for this particular package.